experiments/AMIP/moist_mpi_earth/coupler_driver.jl

```diff
@@ -0,0 +1,191 @@
+# coupler_driver
+# don't forget to run with threading: julia --project --threads 8 (MPI not that useful for debugging coarse runs)
+
+# import packages
+using Pkg
+#Pkg.rm("ClimaAtmos")
+import SciMLBase: step!
+using OrdinaryDiffEq
+using OrdinaryDiffEq: ODEProblem, solve, SSPRK33, savevalues!, Euler
+using LinearAlgebra
+import Test: @test
+using ClimaCore.Utilities: half, PlusHalf
+
+Pkg.add(PackageSpec(name = "ClimaCore", version = "0.10.3"))
+
+# Get the paths to the necessary data files - land sea mask, sst map, sea ice concentration
+include("artifacts.jl")
+
+# import coupler utils
+include("coupler_utils/flux_calculator.jl")
+include("coupler_utils/conservation_checker.jl")
+include("coupler_utils/regridder.jl")
+include("coupler_utils/masker.jl")
+include("coupler_utils/general_helper.jl")
+
+# # initiate spatial and temporal info
+debug_mode = true
+t_end = debug_mode ? 100e2 : 2592000
+tspan = (0, t_end)
+Δt_cpl = 2e2
+saveat = debug_mode ? Δt_cpl * 1 : Δt_cpl * 100
+
+@show debug_mode
+
+# init MPI
+include("mpi/mpi_init.jl")
+
+# init atmos model component
+include("atmos/atmos_init.jl")
+atmos_sim = atmos_init(FT, Y, spaces, integrator, params = params);
+
+# init a 2D bounary space at the surface, assuming the same instance (and MPI distribution if applicable) as the atmos domain above
+boundary_space = ClimaCore.Fields.level(atmos_sim.domain.face_space, half) # global surface grid
+
+# init land-sea mask
+
+mask = LandSeaMask(FT, mask_data, "LSMASK", boundary_space)
+mask = swap_space!(mask, boundary_space) # needed if we are reading from previous run
+
+# init surface (slab) model components
+# we need some types that are defined in these files
+include("slab/slab_utils.jl")
+Pkg.add(url = "https://github.com/CliMA/ClimaLSM.jl")
+include("bucket/bucket_init.jl")
+include("slab/slab_init.jl")
+include("slab_ocean/slab_init.jl")
+include("slab_ice/slab_init.jl")
+
+land_sim = "bucket"
+if land_sim == "slab"
+    slab_sim = slab_init(FT; tspan, dt = Δt_cpl, space = boundary_space, saveat = saveat, mask = mask)
+end
+if land_sim == "bucket"
+    slab_sim = bucket_init(FT;tspan =  FT.(tspan), dt = FT(Δt_cpl), space = boundary_space, saveat = FT(saveat))
+end
+
+prescribed_sst = false
+if prescribed_sst
+    println("No ocean sim - do not expect energy conservation")
+    SST =
+        ncreader_rll_to_cgll_from_space(FT, time_slice_ncfile(sst_data), "SST", boundary_space, outfile = "sst_cgll.nc")  # a sample SST field
+
+    SST = swap_space!(SST, axes(mask)) .* (abs.(mask .- 1)) .+ FT(273.15)
```

Comment:
no need to zero out the SST and SIC fields here, since we account for the masks properly when computing fluxes